File: 06ros/ws02/first_ws/src/topic_pkg/src/subscriber.cpp
//
// Created by zcb on 5/3/21.
//

#include <ros/ros.h>
#include <iostream>
#include <std_msgs/String.h>

void callback(const std_msgs::String::ConstPtr &msg) {
    std::cout << msg->data << std::endl;

}


int main(int argc, char **argv) {
    std::string nodeName = "subscriber_node";
    ros::init(argc, argv, nodeName);

    ros::NodeHandle node;

    std::string topicName = "/zcb01/topic";
    ros::Subscriber subscriber = node.subscribe(topicName, 1000,callback);

    ros::spin();


    return 0;
}

File: 06ros/ws02/first_ws/src/topic_pkg/scripts/subscriber.py
#! /usr/bin/env python
# coding=utf8

import rospy
from std_msgs.msg import String


def callback(msg):
    if not isinstance(msg, String):
        return

    print msg.data


if __name__ == '__main__':
    nodeName = "subscriber_node"
    rospy.init_node(nodeName)

    topicName = "/zcb01/topic"
    rospy.Subscriber(topicName, String, callback=callback, queue_size=1000)

    rospy.spin()
